README_dev.md

```diff
@@ -1,18 +1,56 @@
-# Setting up Development Environment
+# Setting up a Development Environment
 
-These are the instructions for setting up the development dependencies and conforming to the style guide for the Mage.ai repo.
+We'd love to have your contribution, but first you'll need to configure your local environment first!
+
+
+These are the steps:
+
+1. Build Mage image (requires Docker)
+2. Configure virtual environment
+3. Install Git hooks
+4. Install pre-commit hooks
+5. Run dev instance!
+
+First, if you haven't already, clone the repo and navigate to the folder:
+
+```bash
+git clone https://github.com/mage-ai/mage-ai mage-ai
+cd mage-ai
+```
+
+Next, we'll configure our virtual environment.
 
 ## Virtual Environment
-The Mage.ai uses at least Python 3.6 (as per `setup.py`), but the development dependencies will complain if you're not using at least Python 3.8.
 
-As such, make sure you have Python 3.8 on your PATH. Verify this with:
+Mage requires >= Python 3.8. We recommend using `virtualenv` for your new development environment. You can read about how to install and use `virtualenv` [here](https://www.dataquest.io/blog/a-complete-guide-to-python-virtual-environments/).
+
+### Option 1: Virtualenv
+
+Make sure you're in your development repo— `cd mage-ai` and create a new environment:
+
 ```bash
-python --version
+pyenv virutalenv 3.10.0 mage-dev
```

Comment:
I fail to run this command
<img width="297" alt="image" src="https://github.com/mage-ai/mage-ai/assets/80284865/94419f14-2135-4c5e-ba7a-56d170ccb4e4">

Reply:
It's `pyenv shell 3.10`.

Reply:
You're right. I make these changes once https://github.com/mage-ai/mage-ai/pull/3820 is merged— a contributor I was working with _also_ rewrote the instructions and I like his edits better.